cli/module_registry.go

```diff
@@ -0,0 +1,131 @@
+	URL         string            `json:"url"`
+	Description string            `json:"description"`
+	Models      []ModuleComponent `json:"models"`
+	Entrypoint  string            `json:"entrypoint"`
+}
+
+const (
+	defaultManifestFilename = "meta.json"
+)
+
+// CreateModuleCommand runs the command to create a module
+// This includes both a gRPC call to register the module on app.viam.com and creating the manifest file.
+func CreateModuleCommand(c *cli.Context) error {
+	moduleNameArg := c.String("name")
+	orgIDArg := c.String("org_id")
+	publicNamespaceArg := c.String("public_namespace")
+
+	client, err := NewAppClient(c)
+	if err != nil {
+		return err
+	}
+	publicNamespace, err := resolvePublicNamespace(client, orgIDArg, publicNamespaceArg)
+	if err != nil {
+		return err
+	}
+
+	cwd, err := os.Getwd()
+	if err != nil {
+		return errors.Wrap(err, "failed to find the current directory")
+	}
+	// Check to make sure the user doesn't accidentally overwrite a module manifest
+	manifestFilepath := filepath.Join(cwd, defaultManifestFilename)
+	if _, err := os.Stat(manifestFilepath); err == nil {
+		return errors.Errorf("Another module's %v already exists in the current directory. Delete it and try again", defaultManifestFilename)
+	}
+
+	response, err := client.CreateModule(moduleNameArg, publicNamespace)
+	if err != nil {
+		return errors.Wrap(err, "failed to register the module on app.viam.com")
+	}
+	if response.GetModuleId() == "" {
+		return errors.New("empty module id returned. Please report this")
+	}
+
+	fmt.Fprintf(c.App.Writer, "Successfully created '%s'.\n", response.GetModuleId())
+	if response.GetUrl() != "" {
+		fmt.Fprintf(c.App.Writer, "You can view it here: %s \n", response.GetUrl())
+	}
+
+	emptyManifest := ModuleManifest{
+		Name:       response.GetModuleId(),
+		Visibility: ModuleVisibilityPrivate,
+		// This is done so that the json has an empty example
+		Models: []ModuleComponent{
+			{},
+		},
+	}
+	emptyManifestBytes, err := json.MarshalIndent(emptyManifest, "", "  ")
+	if err != nil {
+		return err
+	}
+	//nolint:gosec
+	manifestFile, err := os.Create(manifestFilepath)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create %s", defaultManifestFilename)
+	}
+	if _, err := manifestFile.Write(emptyManifestBytes); err != nil {
+		return errors.Wrapf(err, "failed to write manifest to %s", defaultManifestFilename)
+	}
+	fmt.Fprintf(c.App.Writer, "Configuration for the module has been written to %s\n", defaultManifestFilename)
+	return nil
+}
+
+// resolvePublicNamespace accepts either an orgID or a publicNamespace (one must be an empty string).
+// If publicNamespace is an empty string, it will use the orgID to resolve it.
+func resolvePublicNamespace(client *AppClient, orgID, publicNamespace string) (string, error) {
+	if publicNamespace != "" {
+		if orgID != "" {
+			return "", errors.New("cannot specify both org id and public namespace")
+		}
+		return publicNamespace, nil
```

Comment:
2 part question here:

1. I'm curious how some of the messages appear when additional backend logic returns errors. For example what does it look like when a user enters a `public_namespace` that doesn't exist?
2. Since we have the orgs available through loadOrganizations() should we at this level return an error when the namespace is not found within a user's orgs? Depends on what #1 looks like, because I think we should try to reduce duplicate logic between the CLI and backend.

Reply:
1. 
```
➜  cli git:(zp/2129) ✗ go run viam/main.go module create --public_namespace chicken --name hello-modu
2023/07/07 15:06:40 failed to register the module on app.viam.com: rpc error: code = InvalidArgument desc = no organization found with namespace chicken.
```

Reply:
I think its fine, developers can understand this which is our target audience for the CLI. We can get fancier later.